lib/submit-analytics.js

```diff
@@ -33,50 +120,163 @@ function sha256(data) {
     return hash.digest('hex');
 }
 
-if (isAnalyticsDisabled()) {
-    module.exports = function(){};
-} else {
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {[WebhookProdDetails|WebhookStageDetails|MixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(requestFactory, payload) {
+    const https = require('https');
+
+    return new Promise((resolve, reject) => {
+        const webhookRequest = new requestFactory().buildRequest(payload);
+        var req = https.get(webhookRequest, function (result) {
+            result.on('data', (data) => {
+                // we don't log the data that is returned -- we're just interested in the header
+            });
+        });
+
+        req.on('response', response => {
+            response.on('end', () => {
+                resolve(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: '',
+                    });
+                return;
+            });
+
+            response.on('error', error => {
+                reject(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: error && error.message ? error.message : error,
+                    });
```

Comment:
It is best practice to reject a promise with an instance of `Error` (or a subclass thereof), since that also ensures that a stacktrace captured making it easier to debug.
```suggestion
                const message = error && error.message ? error.message : error;
                const err = new Error(`Failed to dispatch analytics: ${message}`);
                reject(err);
```